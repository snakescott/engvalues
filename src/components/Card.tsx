import { Component } from 'react';
import * as React from 'react';

import './Card.css';

class Card extends Component<any, any> {
  constructor(props: any, context: any) {
    super(props, context);
    this.state = {};
  }

  render() {
    return (
      <div
        className={'Card ' + (this.props.mark || '')}
        style={this.props.style}
        onClick={this.props.onClick}
        >
        { this.props.text }
      </div>
    );
  }
}

namespace Card {
  export const defaultProps = {}
}

export default Card;
