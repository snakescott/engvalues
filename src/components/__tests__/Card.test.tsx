import * as React from 'react';
import * as ReactDOM from 'react-dom';
import Card from '../Card';
import * as renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Card />,
  div);
});

it('renders text', () => {
  const component = renderer.create(
    <Card text="MAXIMUM FLUBSTEP"/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
