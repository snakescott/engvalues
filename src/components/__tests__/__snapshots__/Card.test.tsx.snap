exports[`test renders text 1`] = `
<div
  className="Card "
  onClick={undefined}
  style={undefined}>
  MAXIMUM FLUBSTEP
</div>
`;
